precompiles/assets/tx.go

```diff
@@ -136,16 +139,38 @@ func (p Precompile) RegisterOrUpdateTokens(
 
 	// the price feed must exist
 	_, assetID := assetstypes.GetStakeIDAndAssetIDFromStr(asset.LayerZeroChainID, "", asset.Address)
-	if _, err := p.assetsKeeper.GetSpecifiedAssetsPrice(ctx, assetID); err != nil {
-		return nil, err
+
+	updated := false
+	stakingAsset, _ := p.assetsKeeper.GetStakingAssetInfo(ctx, assetID)
+	if stakingAsset != nil {
+		// this is for update
+		if asset.TotalSupply.IsPositive() {
+			stakingAsset.AssetBasicInfo.TotalSupply = asset.TotalSupply
+		}
+		if len(asset.MetaInfo) > 0 {
+			stakingAsset.AssetBasicInfo.MetaInfo = asset.MetaInfo
+		}
+		updated = true
+	} else {
+		stakingAsset = &assetstypes.StakingAssetInfo{
+			AssetBasicInfo:     &asset,
+			StakingTotalAmount: sdkmath.NewInt(0),
+		}
+
+		// parse oracle info from metaInfo
+		// MetaInfo: `...oracle_info:{chain:Bitcoin,token:BTC,decimal:8,interval:50,contract:0x}...`
+		oracleInfo := oracleInfoMatcher.FindStringSubmatch(asset.MetaInfo)
```

Comment:
I don't love this approach; can we add a new field for `x/oracle` during the registration instead?

Reply:
sure, a new field would be better

Reply:
updated